src/poller/mod.rs

```diff
@@ -0,0 +1,184 @@
+use crate::{
+    audit::log_sync,
+    common::get_unix_time_seconds,
+    db::{delete, get_all},
+    erc20::ERC20Token,
+    gateway::PaymentGateway,
+    transfers::{errors::TransferError, gas_transfers::transfer_gas_to_treasury},
+    types::Invoice,
+};
+
+use alloy::{
+    primitives::Uint,
+    providers::{Provider, RootProvider},
+    rpc::types::eth::TransactionReceipt,
+    transports::http::Http,
+};
+use reqwest::Client;
+use sled::Tree;
+
+/// Checks if a specific token of a specific amount has been received
+/// at a certain address.
+async fn check_if_token_received(
+    token: ERC20Token,
+    invoice: Invoice,
```

Comment:
accept by ref, it reduces unnecessary clones

Reply:
e889da77a756abf1d3390a98806640e31f58cfb7